content/articles/2017-08-purrr-0.2.3.Rmd

```diff
@@ -0,0 +1,229 @@
+---
+title: purrr 0.2.3
+slug: purrr-0.2.3
+description: >
+  A new version of purrr is on CRAN! It features a new family of
+  generic mappers, a tool for tidy plucking of deep data structures,
+  and many other features and fixes.
+author: Lionel Henry
+date: 2018-08-09
+photo:
+  url: https://unsplash.com/photos/TAN1KpDS7Rg
+  author: Erika Lanpher
+categories: [packages]
+---
+
+```{r setup, include = FALSE}
+library("purrr")
+knitr::opts_chunk$set(collapse = TRUE, comment = "#>")
+```
+
+We are pleased to announce that purrr 0.2.3 is now on CRAN! Despite
+the small increment in the version number (our mistake, it should have
+been 0.3.0), this is a rather major release that includes many new
+features and bug fixes.
+
+Install the latest version of purrr with:
+
+```{r, eval = FALSE}
+install.packages("purrr")
+```
+
+You can read about the complete set of changes
+at <https://github.com/tidyverse/purrr/releases/tag/v0.2.3>. In this
+blog post, I will present the highlights of this version which are a
+family of generic mappers, a new `pluck()` function for accessing deep
+elements in a readable way, and a few nifty features and improvements.
+
+
+### Generic mapping
+
+The new `modify` family of functions introduces genericity to mapping.
+This means it is now easier to map functions over vector-like S3
+objects. The reason is twofold:
+
+* `map()` and friends apply functions on the elements of the
+  underlying data structure but that is not always appropriate. An
+  object sometimes contains metadata. A user of such a class probably
+  wants to map over the *elements* of the data of interest rather than
+  over the metadata fields.
+
+* `map()` is type-stable and always returns a list. If you're mapping
+  over an object, chances are you want a similar object back.
+
+`modify()` and its variants solve both these problems via S3
+dispatch. For instance, let's try a conditional map over a data
+frame. Since `map_if()` is type-stable, it returns a list and we lose
+the data frame information:
+
+```{r}
+iris %>% map_if(is.factor, as.character) %>% str()
+```
+
+On the other hand, `modify_if()` is generic and returns a data frame:
+
+```{r}
+iris %>% modify_if(is.factor, as.character) %>% str()
+```
+
+As a developer there are two ways to make your class compatible with
+`modify()` . The easiest is to implement a method for the
+subset-assign operator `[<-` which should be sufficient in most
+cases. Alternatively you can implement methods for the `modify`
+functions themselves as they are all generics.
+
```

Comment:
I used this example:

```r
map(mtcars, unique)
```

because the variable size output seemed more to the point than avoiding recycling. Would you like the recycling example as well?